Source/Core/CylinderGeometry.js

```diff
@@ -80,18 +80,15 @@ define([
         var slices = defaultValue(options.slices, 128);
 
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(length) || length <= 0) {
+        if (!defined(length) || length < 0) {
```

Comment:
Get rid of the `length < 0` check, it's handled in `CylinderGeometry.createGeometry`